### ISSUE TITLE
Sourcemaps not working

### ISSUE THREAD

**Issue by amanitequeen** [getsentry/sentry-cordova#138]
I am working on a Ionic 3 application. The issue interface of Sentry doesn't show the context of js errors, I still have the ununderstandable errors in "1.js". 
- I have uploaded the sourcemaps, they have the "~/" prefix, 
- they appear in my release in "artifact", 
- they have the right "dist" (i tried without and with a "dist"),
- the issue is rightfully connected to the release (if I go to the release interface and see the the list of the issues, my issue appears in there)
- the maps are correct (i didn't check all of them but the generation at app in www/build/ worked)

I am sending my sourcemaps file with 
`sentry-cli releases files blablabalbal upload-sourcemaps www/build/ --rewrite`

what am I missing ?

![1](https://user-images.githubusercontent.com/20946809/60190138-7037c280-9832-11e9-9b02-fcd3fb0fe8dc.png)

the issue stacktrace
![2](https://user-images.githubusercontent.com/20946809/60190140-7037c280-9832-11e9-83a9-5ba556297791.png)

the files in my release artifact
![3](https://user-images.githubusercontent.com/20946809/60190141-7037c280-9832-11e9-9505-2299bcd6fbf2.png)

**Comment by eimermusic** [comment]
Using Cordova, I have a similar issue that I tracked down and worked around. Maybe it helps you figure out your problem.

- Source maps were being uploaded using hashed release ids.
- My errors, however, were coming in with a "proper" application version as the release id. The hash was not picked up properly.

What I did was:

- I verified that the hashed release version was in fact attached to the window object. Look in `platforms/ios/www/index.html`. (could of-course also be android platform for example)
- I manually set the release to that hash value when I initialize Sentry using `Sentry.setRelease(window.SENTRY_RELEASE && window.SENTRY_RELEASE.id);`

````
    // in onDeviceReady...
    const Sentry = cordova.require("sentry-cordova.Sentry");
    Sentry.init({ dsn: 'https://abc123abc123abc123abc123@sentry.io/123456' });
    Sentry.setRelease(window.SENTRY_RELEASE && window.SENTRY_RELEASE.id);
````

I did not figure out why Sentry did not pick up on the release hash, though.

Hope that helps in some way.

**Comment by amanitequeen** [comment]
Hello, Thanks for your response. My release is already set in the sentry init, and since I use the sentry-cordova plugin, i don't have any sentry related code in index.html. Thanks anyway for your help !

**Comment by yurik94** [comment]
@amanitequeen same problem here.
Anyway I found `SENTRY_RELEASE `in /src/index.html but still no `main.js` source map recognized

`sentry-cordova@0.15.0`

**Comment by yurik94** [comment]
> Hello, Thanks for your response. My release is already set in the sentry init, and since I use the sentry-cordova plugin, i don't have any sentry related code in index.html. Thanks anyway for your help !

Resolved!
Check out #61 
I had to remove `release` from `Sentry.init()` and `Sentry.setRelease`

**Comment by mburger81** [comment]
Solution #61 seems to not work for me.
@amanitequeen did you resolve the problem? 

I'm trying to use this on ionic4 and capacitor

**Comment by thmclellan** [comment]
@mburger81 In case it helps, I got source maps working with Ionic 4 and capacitor with the following: 

- Using release parameter as part of sentry.init() 
- My angular.json production section has source maps enabled as follows: 
```
 "sourceMap": {
                "hidden": false,
                "scripts": true,
                "styles": true
              }, 
```
- I'm using Angular CLI for builds with `ng build --prod --aot --output-path=dist/client`
- I found it easier to use sentry-cli than the webpack plugin, and so after the build I'm running the following: 
```
 sentry-cli releases new $RELEASE
 sentry-cli releases files $RELEASE upload-sourcemaps ./dist/client -x .js -x .map --validate --verbose --rewrite --strip-common-prefix
 sentry-cli releases finalize $RELEASE
```
- I'm deleting all map files from the /dist folder before generating Android / iOS builds so they're not embedded in the APK / archive.  `find ./dist -name "*.map" -type f -delete`

The --validate and --verbose options were really helpful to troubleshoot how Sentry was processing the source maps. I got most of the above steps from a great article at https://medium.com/angular-athens/make-angulars-source-code-available-to-sentry-with-gitlab-ci-b3020bd60ae6.

**Comment by michelepatrassi** [comment]
@thmclellan thanks for the snippets! In the article I saw that you're using sentry/browser and not sentry-cordova, does that make any difference?

I followed https://docs.sentry.io/platforms/javascript/ionic/ to set everything up but not all of my sourcemaps are uploaded (getting error like "no response from server") so I cannot map the errors.

**Comment by thmclellan** [comment]
@michelepatrassi Since running on Ionic 4 with Capacitor instead of Cordova, I actually followed the instructions for Angular https://docs.sentry.io/platforms/javascript/angular/ with sentry/browser and didn't use the Cordova version. That covered our needs for crash reporting within the web view. 

For native reporting, since we're on Capacitor, I followed the instructions for native iOS and native Android/Java and made the changes directly in Xcode and Android studio. Sorry no easy answers on above error, may be specific to Cordova.
